### PR TITLE
add mgmt cluster description to prov cluster description as fallback in home cluster list and main side nav clusters

### LINKED ISSUE #10441: Add cluster description to cluster side menu and main menu
Internal reference: SURE-6691
Requested in 2.6.13

A user wants to add the cluster description to the Cluster Side menu and the main cluster menu. They have added some unique cluster information to the description field which is only visible when you open the full cluster view. It would become handy to have the cluster description already available in the cluster side menu without opening cluster by cluster to view and find the correct one.

Actual behavior:
No possible to see the description field in the burger menu.

Expected behavior:
Possible to read the description field directly in the burger menu, with no need to open the full cluster view.

@kwwii and @edenhernandez-suse have come up with an appropriate design: https://xd.adobe.com/view/c1936434-760f-44a6-8f7a-788b16f50722-94c0/?fullscreen

Note: Consider long descriptions and truncation

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #10441 (based on reopening from QA by comment https://github.com/rancher/dashboard/issues/10441#issuecomment-2047009103)

### Occurred changes and/or fixed issues

- add `mgmt` cluster description to `prov` cluster description as fallback in home cluster list and main side nav clusters

### Technical notes summary

Descriptions were not appearing because both lists are based on CAPI clusters and the ones mentioned by QA are MGMT clusters. Added the MGMT clusters to the home cluster list and main side nav clusters

### Areas or cases that should be tested

- Edit `local` cluster and add a description
- Provision a AKS, EKS or GKE cluster with a description
- make sure description appears in the home page cluster list and main side nav clusters

### Screenshot/Video

Screenshot 2024-04-10 at 16 20 33
Screenshot 2024-04-10 at 16 27 26

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/pages/home.vue [hunk 1]
```diff
@@ -91,6 +91,10 @@ export default {
     ...mapGetters(['currentCluster', 'defaultClusterId', 'releaseNotesUrl']),
     mcm: mapFeature(MULTI_CLUSTER),
 
+    mgmtClusters() {
+      return this.$store.getters['management/all'](MANAGEMENT.CLUSTER);
```
Comment (momesgin) [inline shell/pages/home.vue:95]: I'm curious how the unit test in `TopLevelMenu.test.ts` didn't catch this case? is it possible to create a test case to cover both scenarios(i.e. `pCluster?.description || x.description`)?

Comment (aalves08) [reply shell/pages/home.vue:95]: Sure thing. Unit test added.

File: shell/components/nav/__tests__/TopLevelMenu.test.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import TopLevelMenu from '@shell/components/nav/TopLevelMenu';
 import { SETTING } from '@shell/config/settings';
 import { mount, Wrapper } from '@vue/test-utils';
 
-// DISCLAIMER: This should not be added here, although we have several store requests which are irrelevant
+// DISCLAIMER: This should not be added here, although we have several store requests which are irrelevant//
```
Comment (momesgin) [inline shell/components/nav/__tests__/TopLevelMenu.test.ts:5]: I think you forgot to remove this 🙈

Comment (aalves08) [reply shell/components/nav/__tests__/TopLevelMenu.test.ts]: 🤦 fixed

**Review by momesgin — APPROVED** [review]
I tested with an AKS cluster and LGTM

**Comment by aalves08** [comment]
@izaac @yonasberhe23 I've added some coverage for this under the RKE1 amazon EC2 cluster provisioning. I did my best to test this fully but it needs a remote instance (couldn't test against this branch env) so the only I didn't test was reading the description from the home page cluster list, but that has been covered previously so I don't think it will put us into any problems. Just for FYI if you notice anything weird (Idk if this test in particular runs on our pipeline or not). 🙏

**Pull request merged**